### PR TITLE
WIP: implement DXdao token list fetching

### PR DESCRIPTION
**Pull request by luzzif**

### CODE REVIEW

**Comment by AugustoL** [comment]
@luzzif ready for merge?

File: dist/constants.d.ts [hunk 1]
```diff
@@ -21,6 +22,10 @@ export declare const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000"
 export declare const FACTORY_ADDRESS: {
     [chainId: number]: string;
 };
+export declare const TOKEN_REGISTRY_ADDRESS: {
+    [chainId: number]: string;
+};
+export declare const DXSWAP_TOKEN_LIST_ID = 5;
```
Comment (AugustoL) [inline dist/constants.d.ts:28]: this shouldnt be by chainId too?

**Comment by AugustoL** [comment]
@luzzif you can the remove the commit adding the dist folder, we will update it from develop to master.

**Comment by luzzif** [comment]
> @luzzif you can the remove the commit adding the dist folder, we will update it from develop to master.

Is it a problem to keep it? Plus, I feel like it's really not the best way to operate. We should probably exclude the `dist` folder from the library's codebase and generate it when publishing to some sort of package manager like `npm`. If we want to keep a higher level of decentralization, we should just add a `postinstall` npm script that performs the transpilation from TS to JS strictly when needed and that's it, no need for a package manager. Even though if it is for decentralization purposes, having the repo on Github is probably not the best solution anyway.

**Comment by AugustoL** [comment]
Yep, thats how we will do it in next release :grimacing:, thats the plan. This is not a good way to do it but the easiest one for now, thats why we are keeping track of it in github. 
The idea is to do a final v1 release with the mainnet deploy and package manager configured hopefully after this one.

**Comment by luzzif** [comment]
@AugustoL should have done everything. I've made the token list id network-dependent (event though I currently don't know the actual list id on Rinkeby or any other network outside of mainnet) and reverted the dist folder so that it's not an issue. Feel free to merge if everything's ok to you

**Pull request merged**